src/Client.ts

```diff
@@ -5,22 +5,32 @@ import {
 } from '@kontent-ai/delivery-sdk';
 import packageInfo from '../package.json';
 import { selectedProjectCookieName } from './const';
-import { defaultProjectId } from './Utilities/SelectedProject';
+import validator from 'validator';
 
 const sourceTrackingHeaderName = 'X-KC-SOURCE';
 
-// environment variables
-const projectId = process.env.REACT_APP_PROJECT_ID || '';
 const previewApiKey = process.env.REACT_APP_PREVIEW_API_KEY || '';
 
 const cookies = new Cookies(document.cookie);
 
-let currentProjectId = projectId || cookies.get(selectedProjectCookieName);
-if (currentProjectId) {
-  cookies.set(selectedProjectCookieName, currentProjectId, { path: '/' });
-} else {
-  currentProjectId = defaultProjectId;
-}
+const getProjectId = (): string => {
+  const projectId = process.env.REACT_APP_PROJECT_ID || '';
+
+  let currentProjectId = projectId;
+
+  if (projectId) {
+    if (!validator.isUUID(projectId)) {
+      console.error(
+        `Your projectId (${projectId}) given in your environment variables is not a valid GUID.`
+      );
+    }
+    return projectId;
+  }
+  currentProjectId = cookies.get(selectedProjectCookieName);
+  return currentProjectId ? currentProjectId : '';

```

Comment:
Good job refactoring this code into a function. Let's simplify the function a bit. The `let` variable is not really necessary. You want to check whether there is a valid `projectId` in the environment variable and alternatively fall back to the cookie and the fall back to an empty string. Something like this might work.
```ts
const projectIdFromEnv = process.env.VAR_NAME || '';

if (projectIdFromEnv && !validator.isUUID(projectIdFromEnv)) {
  // ...
}

return projectIdFromEnv ?? cookies.get(...) ?? '';
```
What do you think?